packages/shared/src/utils.ts

```diff
@@ -89,16 +89,18 @@ export function getNewBounds(oldBounds: Bounds, value: number): Bounds {
 }
 
 export function getBounds(
-  valuesArray: NdArray<number[]> | number[],
-  errorArray?: NdArray<number[]> | number[]
+  valuesArray: AnyNumArray,
+  errorArray?: AnyNumArray
 ): Bounds | undefined {
   assertDataLength(errorArray, valuesArray, 'error');
 
-  const values = toArray(valuesArray);
-  const errors = errorArray && toArray(errorArray);
+  const values = getValues(valuesArray);
+  const errors = errorArray && getValues(errorArray);
 
-  const bounds = values.reduce<Bounds>(
-    (acc, val, i) => {
+  // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error
+  // @ts-ignore (https://github.com/microsoft/TypeScript/issues/44593)
+  const bounds = values.reduce(
+    (acc: Bounds, val: number, i: number) => {
```

Comment:
`values` has type `NumArray` (`TypedArray | number[]`), and that's too hard for `reduce` to handle in TS -- but it runs perfectly fine since both typed and plain arrays have a `reduce` method.